### PR TITLE
everdo: init at 1.9.0

### LINKED ISSUE #311917: Package request: everdo
**Project description**
A powerful cross-platform GTD (Getting Things Done) app.

**Metadata**

* homepage URL: https://everdo.net/
* source URL: not open source
* license:
* platforms: linux, windows, macos

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes  #311917

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#everdo
> tree result/
 result
└──  bin
   └──  everdo
```

File: pkgs/by-name/ev/everdo/package.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  lib,
+  fetchurl,
+}:
+appimageTools.wrapType2 rec {
+  pname = "everdo";
+  version = "1.9.0";
+
+  src = fetchurl {
+    url = "https://release.everdo.net/${version}/Everdo-${version}.AppImage";
+    sha256 = "sha256-0yxAzM+qmgm4E726QDYS9QwMdp6dUcuvjZzWYEZx7kU=";
```
Comment (eclairevoyant) [inline pkgs/by-name/ev/everdo/package.nix:12]:
```suggestion
    hash = "sha256-0yxAzM+qmgm4E726QDYS9QwMdp6dUcuvjZzWYEZx7kU=";
```

**Review by eclairevoyant — COMMENTED** [review]
Not tested this yet, but a couple points of feedback:

**Comment by luftmensch-luftmensch** [comment]
@eclairevoyant applied suggestion.

**Review by chewblacka — CHANGES_REQUESTED** [review]
You've got your `everdo` and `proguard` packages mixed into the same commit.

**Review by chewblacka — COMMENTED** [review]
Also given it's a gui app, a desktop file and icon would be a good idea if possible.

**Comment by luftmensch-luftmensch** [comment]
> You've got your `everdo` and `proguard` packages mixed into the same commit.

Ops. I've missed that

**Comment by chewblacka** [comment]
For icon: `https://archive.org/download/everdo/everdo.png`

For desktop file:

    desktopItem = makeDesktopItem {
      type = "Application";
      name = "everdo";
      desktopName = "Everdo";
      comment = "A productivity app for GTD (Getting Things Done)";
      icon = "everdo";
      exec = "everdo %u";
      categories = [ "Office" ];
    };

**Comment by luftmensch-luftmensch** [comment]
@chewblacka Since the App image already had a desktop item & an icon I patched using the `extraInstallCommands` step. 

Let me know if it's ok, or if I should ditch completely and using the `desktopItem` options

Additional infos:
```
> nix build .#everdo
> tree result/
 result
├──  bin
│  └──  everdo -> /nix/store/fjg0i7k0vhyda62lcvb4wdh302h9p2wg-everdo-1.9.0-bwrap
└──  share
   ├──  applications
   │  └──  everdo.desktop
   └──  icons
      └──  hicolor
         ├──  32x32
         │  └──  apps
         │     └──  everdo.png
         ├──  64x64
         │  └──  apps
         │     └──  everdo.png
         ├──  128x128
         │  └──  apps
         │     └──  everdo.png
         ├──  256x256
         │  └──  apps
         │     └──  everdo.png
         ├──  512x512
         │  └──  apps
         │     └──  everdo.png
         └──  1024x1024
            └──  apps
               └──  everdo.png
```

**Comment by chewblacka** [comment]
That looks great! Just one minor addition, I would change the `--replace` to `--replace-fail` in the `substituteInPlace` just because the former is deprecated and gives the warning:

`WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}.`

**Comment by luftmensch-luftmensch** [comment]
> That looks great! Just one minor addition, I would change the `--replace` to `--replace-fail` in the `substituteInPlace` just because the former is deprecated and gives the warning:
> 
> `WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}.`

Didn't know that.

**Review by chewblacka — APPROVED** [review]
LGTM

**Pull request merged**